# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/peerconnection/TransceiverImpl.cpp
# Commit: 0b03dd9d20ac
# Full Hash: 0b03dd9d20ace235502714101097fe34d1864563
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 1
# Description:
#   Bug 1493613 - Update muted state through MediaStreamTrackSource. r=bwc,smaug
#   
#   This ensures all clones of the original track also receives the new muted state.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37933
# ==============================================================================

diff -r cf84eeffed22 -r 0b03dd9d20ac media/webrtc/signaling/src/peerconnection/TransceiverImpl.cpp
--- a/media/webrtc/signaling/src/peerconnection/TransceiverImpl.cpp	Tue Jul 30 14:55:48 2019 +0000
+++ b/media/webrtc/signaling/src/peerconnection/TransceiverImpl.cpp	Wed Jul 31 09:35:06 2019 +0000
@@ -242,6 +242,15 @@
   return NS_OK;
 }
 
+void TransceiverImpl::SetReceiveTrackMuted(bool aMuted) {
+  if (!mReceiveTrack) {
+    return;
+  }
+
+  // This sets the muted state for mReceiveTrack and all its clones.
+  static_cast<RemoteTrackSource&>(mReceiveTrack->GetSource()).SetMuted(aMuted);
+}
+
 nsresult TransceiverImpl::UpdatePrincipal(nsIPrincipal* aPrincipal) {
   if (mJsepTransceiver->IsStopped()) {
     return NS_OK;